### PR TITLE
dpg/mgmt, upgrade codegen

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
1. upgrade autorest.java version (not the cadl one)
2. add `jacoco.skip` in `mvn package` (change to generated POM to promote best practice of not skipping jacoco)
3. support nested readme in below case (mgmt)

test on mgmt https://github.com/openapi-env-test/azure-rest-api-specs/pull/3641/checks?check_run_id=11457250637

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by haolingdong-msft — APPROVED** [review]

File: eng/mgmt/automation/api-specs.yaml [hunk 1]
```diff
@@ -21,6 +21,10 @@ containerregistry:
   suffix: generated
 containerservice:
   suffix: generated
+containerservice/Microsoft.ContainerService/aks:
```
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply eng/mgmt/automation/api-specs.yaml:24]: Not really. RP is still "containerservice". But they want to separate it to 2 SDKs (aks and fleet).

I will check whether autocent works (maybe in a later PR).

[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply eng/mgmt/automation/api-specs.yaml:24]: > Is there a format rule for RP name? Current autocent.js implementation in mgmt uses this as well..
> 
> https://github.com/Azure/azure-sdk-for-java/blob/4075271d6923e5f17a4b4bb81df8428439e38be5/sdk/resourcemanager/autocent.js#L82
> 
> Can we use the RP name to construct valid specs url something like this?:
> 
> ```js
>   RP.substring(0, RP.indexOf("/")) + "/resourcemanager" + RP.substring(RP.indexOf("/"))
> ```

Yes, we probably could use this.

[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply eng/mgmt/automation/api-specs.yaml:24]: Another TODO is autocent is to remove the libs that already deprecated libs e.g. batchai.

[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply eng/mgmt/automation/api-specs.yaml:24]: issue created: https://github.com/Azure/azure-sdk-for-java/issues/33621

added machinelearningservices

**Pull request merged**